File: server/GarageAPI/Models/ParkingLevel.cs
using Core;
using GarageAPI.Enums;
using System.Collections.Generic;

namespace GarageAPI.Models
{
    public class ParkingLevel : BaseEntity
    {
        public string Code { get; set; }
        public ICollection<ParkingSpace> ParkingSpaces { get; set; } = new HashSet<ParkingSpace>();
        public int AvailableCarSpaces { get; set; }
        public int AvailableMotorbikeSpaces { get; set; }
    }
}

File: client/src/app/modules/garage/classes/parking-level.ts
import { ParkingSpace } from "@modules/garage/classes/parking-space";

export class ParkingLevel {
  id: string;
  code: string;
  order: number;
  parkingSpaces: ParkingSpace[];
  availableCarSpaces: number;
  availableMotorbikeSpaces: number;
}
